### PR TITLE
Fix: MotherDuck token from plugin config should override env var

### PR DESCRIPTION
**Pull request by guenp**
This PR fixes a bug:
- When passing a token via the config, this didn't work because there is a typo (by yours truly 😬) [here](https://github.com/duckdb/dbt-duckdb/blob/8a6e0dc33cfedd580be1b0a0a8e970b6fce05a5f/dbt/adapters/duckdb/plugins/motherduck.py#L16).

The PR also addresses how a user can pass the MotherDuck token to `duckdb.connect`:

- Adds a `motherduck_token` property to `DuckDBCredentials` that fetches it from the plugin config.
- The user can specify the token in the path (e.g. `"md:?motherduck_token=<token>"`. This will override a `motherduck_token` environment variable if it exists.
- We no longer use an explicit `SET` command and instead pass a `config` dictionary to `duckdb.connect`. Any config options passed this way will override the corresponding environment variable as well as any options set in the path.

So the behavior is now the same as in duckdb: env var < path < config. (where left < right means "right overrides left").

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -40,9 +40,7 @@ def _dbt_duckdb_version():
         "dbt-core~=1.7.0",
         "duckdb>=0.7.0",
     ],
-    extras_require={
-        "glue": ["boto3", "mypy-boto3-glue"],
-    },
+    extras_require={"glue": ["boto3", "mypy-boto3-glue"], "md": ["duckdb>=0.7.0,<=0.9.2"]},
```
Comment (guenp) [inline setup.py:43]: I added this so the CI succeeds (MotherDuck currently does not support 0.10).
See run here: https://github.com/motherduckdb/dbt-duckdb/actions/runs/7895023998/job/21546711483

Comment (jwills) [reply setup.py:43]: No worries

File: dbt/adapters/duckdb/credentials.py [hunk 2]
```diff
@@ -155,6 +155,15 @@ def is_motherduck(self):
     def _is_motherduck(scheme: str) -> bool:
         return scheme in {"md", "motherduck"}
 
+    @property
+    def motherduck_token(self) -> str:
```
Comment (jwills) [inline dbt/adapters/duckdb/credentials.py:159]: So I'm a little thrown by this-- I thought the normal place folks put their MD token was in the `path` setting as opposed to by using the MD plugin configuration. There may be more places we want to look for the token besides the plugin config?

Comment (guenp) [reply dbt/adapters/duckdb/credentials.py]: I moved it out of this file entirely :)

**Comment by guenp** [comment]
All checks are green: https://github.com/motherduckdb/dbt-duckdb/actions/runs/7909652885/job/21591671925

After we merge this we should be OK to release! 🎉

File: setup.py (hunk 1 above)
Comment (jwills) [inline setup.py:43]: Do you want me to keep this bit in for now?

**Review by jwills — COMMENTED** [review]
Looks great @guenp -- just one Q and then I will merge it

File: setup.py (hunk 1 above)
Comment (guenp) [reply setup.py:43]: Yes please! I think will will be super helpful to have `pip install dbt-duckdb[md]` as an option ready to go for people to use before we roll out 0.10 support, and I can open a PR to bump this version once that lands

Comment (guenp) [reply setup.py:43]: (We also need to keep this line in tox.ini)
```
-e.[md]
```

**Pull request merged**